datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,63 @@
+package de.adorsys.datasafe.rest.impl.security;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JwtAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private final AuthenticationManager authenticationManager;
+
+    public JwtAuthenticationFilter(AuthenticationManager authenticationManager) {
+        this.authenticationManager = authenticationManager;
+
+        setFilterProcessesUrl(SecurityConstants.AUTH_LOGIN_URL);
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request,
```

Comment:
It is much better to add specialized, publicly exposed endpoint for login.
This method has several issues - getting credentials from request parameters instead of header and supporting any kind of HTTP method - you can use GET,POST,PUT to authenticate. Please use POST endpoint for authentication in the separate controller.
Refer to 
https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/REST_Security_Cheat_Sheet.md